src/main/java/com/aws/greengrass/clientdevices/auth/api/ClientDevicesAuthServiceApi.java

```diff
@@ -49,8 +54,12 @@ public ClientDevicesAuthServiceApi(SessionManager sessionManager, DeviceAuthClie
     public boolean verifyClientDeviceIdentity(String certificatePem) {
         // Allow internal clients to verify their identities
         if (deviceAuthClient.isGreengrassComponent(certificatePem)) {
+            domainEvents.emit(new VerifyClientDeviceIdentityEvent(
+                    VerifyClientDeviceIdentityEvent.VerificationStatus.SUCCESS));
             return true;
         } else {
+            domainEvents.emit(new VerifyClientDeviceIdentityEvent(
```

Comment:
i'm not clear on why we determine success/failure based on `deviceAuthClient.isGreengrassComponent(certificatePem)`, what's the reason?  i would assume we'd want to determine success based on the overall result of this method, but i might be missing something

Reply:
> and then depending on if it returns true or false

what i'm saying is that we're returning success based on `deviceAuthClient.isGreengrassComponent(certificatePem)` rather than the overall result of `verifyClientDeviceIdentity`, doesn't seem like that matches the above explanation?

Reply:
Updated it to track metrics from VerifyIotCertificate.java as well